Python/ast_opt.c

```diff
@@ -971,6 +726,42 @@ astfold_withitem(withitem_ty node_, PyArena *ctx_, _PyASTOptimizeState *state)
     return 1;
 }
 
+static int
+fold_const_match_patterns(expr_ty node, PyArena *ctx_, _PyASTOptimizeState *state)
+{
+    switch (node->kind)
+    {
+        case UnaryOp_kind:
+        {
+            if (node->v.UnaryOp.op == USub &&
+                node->v.UnaryOp.operand->kind == Constant_kind) {
+                PyObject *operand = node->v.UnaryOp.operand->v.Constant.value;
+                PyObject *folded = PyNumber_Negative(operand);
+                return make_const(node, folded, ctx_);
+            }
+            break;
+        }
+        case BinOp_kind:
+        {
+            operator_ty op = node->v.BinOp.op;
+            if ((op == Add || op == Sub) &&
+                node->v.BinOp.right->kind == Constant_kind) {
```

Comment:
```suggestion
                node->v.BinOp.right->kind == Constant_kind)
            {
```

I think we do this when the indent looks weird.